adagrams/game.py

```diff
@@ -1,11 +1,151 @@
+import random
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
 def draw_letters():
-    pass
+    #Create a set of 10 letters that matches the letter frequency distribution#
+    draw = []
+    letter_pool_copy = LETTER_POOL.copy()
+    letters_list = list(letter_pool_copy.keys())
+
+    while len(draw) < 10:       
+        #Take a random number within the len of letter pool and get the letter in that index
+        random_letter = letters_list[random.randint(0,len(letter_pool_copy)-1)]
+        
+        #Check if the letter is still availabe in the letter pool and add it to the hand
+        if letter_pool_copy[random_letter] >= 1:
+            draw.append(random_letter)
+            
+            #reduce the frequency in letter pool, for each letter added to the hand
+            letter_pool_copy[random_letter] -= 1         
+    return draw
 
 def uses_available_letters(word, letter_bank):
-    pass
+    #check if the word is an anagram of some or all of the given letters in the hand
+
+    letter_bank_copy = letter_bank.copy()
+    word = word.upper()  
```

Comment:
Nice work on ensuring that all the letters where of the same casing. You could also use `casefold`, which is used for caseless matching. You can find more information on the method [here](https://docs.python.org/3/library/stdtypes.html#str.casefold).

Also, be mindful of re-assigning the parameter of a function. That could cause issues for the caller of the function if the change is not expected. Usually we want to avoid this unless specifically told to.